pandas/core/arrays/base.py

```diff
@@ -712,7 +722,7 @@ def fillna(self, value=None, method=None, limit=None):
             new_values = self.copy()
         return new_values
 
-    def dropna(self):
+    def dropna(self) -> ExtensionArray:
```

Comment:
```suggestion
    def dropna(self: ExtensionArrayT) -> ExtensionArrayT:
```

and then the cast in SparseArray._reduce can be removed.

but will produce `pandas/core/arrays/base.py:733: error: Incompatible return value type (got "Union[ExtensionArray, Any]", expected "ExtensionArrayT")  [return-value]` since the return type of `ExtensionArray.__getitem__` is `Union[ExtensionArray, Any]` and this needs to be overloaded.

Reply:
I made that change and added overloads on `__getitem__()`